core/src/main/java/io/undertow/server/handlers/encoding/EncodingMapping.java

```diff
@@ -54,6 +54,20 @@ public Predicate getAllowed() {
     }
 
     @Override
+    public boolean equals(Object o) {
```

Comment:
This is wrong, compareTo compares on priority while equals compares on name, so they are inconsistent.